user-agent.js

```diff
@@ -0,0 +1,36 @@
+import { version } from './package.json';
+
+class UserAgent {
+  constructor(client) {
+    if (!client) {
+      throw new Error('"client" arg is required to create a UserAgent.');
+    }
+    this._client = client;
+  }
+
+  toString() {
+    let client = [
+      `Percy/${this._apiVersion()}`,
+      this._client._clientInfo,
```

Comment:
Small, optional change: it feels like `clientInfo` and `environmentInfo` could be public methods on the client object, especially since they're passed in from the user as options to the Client constructor so it's not like we're trying to hide an internal implementation or something like that.

This is mostly motivated to avoid a protected method access here in `client._clientInfo` and instead being able to do `client.clientInfo` and we just expose clientInfo as a more public part of the Client object.